src/Serializer/SerializerContextBuilder.php

```diff
@@ -94,7 +94,7 @@ public function createFromRequest(Request $request, bool $normalization, array $
         }
 
         foreach ($resourceMetadata->getItemOperations() as $operation) {
-            if ('PATCH' === $operation['method'] && \in_array('application/merge-patch+json', $operation['input_formats']['json'] ?? [], true)) {
+            if (\in_array('application/merge-patch+json', $operation['input_formats']['json'] ?? [], true) && 'PATCH' === $operation['method']) {
```

Comment:
```suggestion
            if ('PATCH' === ($operation['method'] ?? '') && \in_array('application/merge-patch+json', $operation['input_formats']['json'] ?? [], true)) {
```

The comparison should be done first cause it's a less complex operation than `in_array`.

Reply:
Ok, it should me pointing at the api-platform:2.5 branch.